### DISCUSSION TITLE
Intergration of Quplexity into mtq-core.

### DISCUSSION THREAD

**Discussion by MrGilli** [munich-quantum-toolkit/core#652] [Ideas]
Hi all,

My name is Jacob Gill, I'm the founder and developer of Quplexity, a free and open source modular tool/library written in x86 and ARM/ARM64 Assembly aimed at providing Quantum Computer/Circuit simulators (QCS) with their essential logic at improved speeds with very little overhead. Because Quplexity is written in Assembly it provides significant and reliable performance benefits over the same functions written in C++ or Python.

I would love the opportunity to integrate the Quplexity library into mqt-core, with the aim of enhancing computational times and the overall "weight" of mqt-core program, I take to gain nothing from doing this, I just want to contribute to the field of quantum computing. To further ensure you of Quplexity's capabilities I have carefully put together a lot of documentation, most notably "A User Guide/Manual" and "Speed/performance comparisons". You should be able to conduct the tests I have done on your own machine with the help of the C++ library "Chrono".

The Quplexity library is split up into two main sections: ./ARM the Quplexity code for ARM CPU's (even raspberry pi's) and ./Intel which holds the Quplexity code for Intel CPU's. Quplexity is also currently being integrated into the Qrack project and I'm in close contact with the founder: Dan Strano.

Thanks, let me know if you guys are ok/interested in me intergrating quplexity into mqt-core.

GitHub: https://github.com/MrGilli/Quplexity

**Comment by burgholzer** [comment]
Hi 👋🏼 

Thanks for the introduction to Quplexity.
As off now, it's functionality seems to be too limited to justify the addition of an external dependency to this project and everything that comes with that.
Furthermore, we ship on a very broad range of machines trying to maximize usability, which we would compromise by adopting Quplexity.
We'll keep an eye out though for the developments in your project and might reconsider at a later point in time.

**Reply by MrGilli** [reply]
Hi 👋🏼,

Thanks for getting back to me. What functionality would you want to see in Quplexity that will justify it's use in MTQ-Core?
In relation to being used on a broad range of machines, Quplexity shouldn't compromise this as it supports all Intel processors and all ARM processors. If their is a machine that doesn't have an intel or ARM CPU you/I could add functionality to not compile the assembly files and run on the traditional C++ code. I would like to add support for more CPU's so if you had any in mind please let me know and I'll get working on it.
